src/Mvc/Mvc.Analyzers/src/Startup/StartupAnalyzer.Diagnostics.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    public partial class StartupAnalzyer : DiagnosticAnalyzer
+    {
+        internal readonly static DiagnosticDescriptor UnsupportedUseMvcWithEndpointRouting = new DiagnosticDescriptor(
+            "MVC1005", 
+            "Cannot use UseMvc with Endpoint Routing.", 
+            "Using '{0}' to configure MVC is not supported while using Endpoint Routing. To continue using '{0}', please set 'MvcOptions.EnableEndpointRounting = false' inside '{1}'.",
```

Comment:
Should this guide the user to use `UseEndpointRouting` instead? All migrating users would see this error and we want to steer them towards endpoint routing. 
Additionally there's a way to do help link with a diagnostic. We should do a placeholder aka.ms link and point it to a migration guide

Reply:
Yes, we should do that.

I think we should possibly include both?